nb-api/src/main/java/io/nosqlbench/api/markdown/aggregator/MutableMarkdown.java

```diff
@@ -69,13 +74,19 @@ private void parseStructure(String rawMarkdown) {
                 } else if (node instanceof WhiteSpace) {
                 } else if (node instanceof YamlFrontMatterBlock) {
                 } else {
-                    throw new RuntimeException("The markdown file at '" + this.path.toString() + "' must have an initial heading as a title, before any other element, but found:" + node.getClass().getSimpleName());
+                    if(this.path != null)
+                        throw new RuntimeException("The markdown file at '" + this.path.toString() + "' must have an initial heading as a title, before any other element, but found:" + node.getClass().getSimpleName());
+                    else
+                        throw new RuntimeException("The markdown string provided must have an initial heading as a title, before any other element, but found: "+ node.getClass().getSimpleName());
                 }
                 node=node.getNext();
             }
         }
         if (frontMatter.getTitle()==null || frontMatter.getTitle().isEmpty()) {
-            throw new RuntimeException("The markdown file at '" + this.path.toString() + "' has no heading to use as a title.");
+            if(this.path != null)
+                throw new RuntimeException("The markdown file at '" + this.path.toString() + "' has no heading to use as a title.");
+            else
+                throw new RuntimeException("The markdown string provided has no heading to use as a title.");
         }
```

Comment:
Ok for now, but let's try to add _some_ context for users in these error messages. They shouldn't have to go fishing for where the issue might be. If there are other outputs in the display context to get them focused on the right files, that's good, but it's always important to put essential troubleshooting context into errors.

Reply:
We could pass a markdown name to the constructor in case we don't have a path to an actual file on the filesystem?

Reply:
Or something like that. Let's try to keep it as simple as possible, but you have the right idea.